cmd/cmd.go

```diff
@@ -1,93 +1,104 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"fmt"
 	"os"
-	"strings"
 	"time"
 
 	"github.com/Clever/batchcli/runner"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/sfn"
 )
 
 func main() {
-	functionCmd := flag.String("cmd", "", "The command to run")
-	localRun := flag.Bool("local", false, "Local mode - auto-assigns random AWS Batch config")
+	activityName := flag.String("name", "", "The activity name to register with AWS Step Functions")
+	activityCmd := flag.String("cmd", "", "The command to run to process activity tasks")
+	sfnRegion := flag.String("region", "", "The AWS region to send Step Function API calls")
 	printVersion := flag.Bool("version", false, "Print the version and exit")
-	resultsLocation := flag.String("results-location", "workflow-results-dev", "name of the table for getting and setting job results")
-	//parseArgs := flag.Bool("parseargs", true, "If false send the job payload directly to the cmd as its first argument without parsing it")
 
 	flag.Parse()
 
 	if *printVersion {
-		fmt.Println(Version)
+		//fmt.Println(Version)
 		os.Exit(0)
 	}
 
-	if *functionCmd == "" {
-		fmt.Println("Required command to execute")
+	if *activityName == "" {
+		fmt.Println("activityname is required")
 		os.Exit(1)
 	}
-
-	var job runner.BatchJob
-	if *localRun {
-		// TODO: allow CLI to set fake job-ids (or inputs)
-		job = runner.NewMockBatchJob([]string{})
-	} else {
-		j, err := runner.NewBatchJobFromEnv()
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		}
-		job = j
-	}
-
-	if *resultsLocation == "" {
-		fmt.Println("-results-location can not be an empty string")
+	if *activityCmd == "" {
+		fmt.Println("cmd is required")
 		os.Exit(1)
 	}
-
-	// TODO: figure out a better way to switch between prod/dev resultsLocations
-	// currently using env var since that easily works across docker containers
-	// which specify `CMD [..]` or `ENTRYPOINT [...]`
-	if strings.Contains(os.Getenv("AWS_BATCH_JQ_NAME"), "production") {
-		*resultsLocation = "workflow-results"
-	}
-
-	store, err := initalizeStore(*localRun, *resultsLocation)
-	if err != nil {
-		fmt.Println(err)
+	if *sfnRegion == "" {
+		fmt.Println("region is required")
 		os.Exit(1)
 	}
 
-	taskRunner, err := runner.NewTaskRunner(*functionCmd, flag.Args(), job, store)
+	ctx := context.Background()
+
+	// register the activity with AWS (it might already exist, which is ok)
+	sfnapi := sfn.New(session.New(), aws.NewConfig().WithRegion(*sfnRegion))
+	createOutput, err := sfnapi.CreateActivityWithContext(ctx, &sfn.CreateActivityInput{
```

Comment:
This would go into our `catapult` as part of deployments of apps. So eventually we should just expect the activity to always exist as a `env--app`

Reply:
what's the reason for putting it in catapult? I think we could do this w/o any changes to catapult...

Reply:
My bad forgot that activities don’t actually have any container definition data

Reply:
Am I right to assume then that you can create a `StateMachine` without actually knowing the exact ARN for the activities it contains?

Reply:
just confirmed that you can create a state machine w/ a non-existent activity, so we're good! 

```
aws --region us-west-2 stepfunctions create-state-machine --name raf-test-state-machine-2 --role-arn arn:aws:iam::589690932525:role/raf-test-step-functions --definition '{
>     "Comment": "Testing out step functions",
>     "StartAt": "raf-test-state",
>     "Version": "1.0",
>     "TimeoutSeconds": 300,
>     "States": {
>         "raf-test-state": {
>             "Resource": "arn:aws:states:us-west-2:589690932525:doesnt-exist-yet",
>             "Type": "Task",
>             "End": true
>         }
>     }
> }'
{
    "creationDate": 1505534549.191, 
    "stateMachineArn": "arn:aws:states:us-west-2:589690932525:stateMachine:raf-test-state-machine-2"
}
```